server/handler/uast.go

```diff
@@ -79,9 +80,52 @@ func Parse(bbblfshServerURL string) RequestProcessFunc {
 	}
 }
 
-// Filter : placeholder method
+type filterRequest struct {
+	Protobufs string `json:"protobufs"`
+	Filter    string `json:"filter"`
+}
+
+// Filter returns a function that filters UAST protobuf and returns UAST JSON
 func Filter() RequestProcessFunc {
 	return func(r *http.Request) (*serializer.Response, error) {
-		return nil, serializer.NewHTTPError(http.StatusNotImplemented, http.StatusText(http.StatusNotImplemented))
+		var req filterRequest
+		body, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			return nil, err
+		}
+
+		err = json.Unmarshal(body, &req)
+		if err != nil {
+			return nil, serializer.NewHTTPError(http.StatusBadRequest, err.Error())
+		}
+
+		data, err := base64.StdEncoding.DecodeString(req.Protobufs)
+		if err != nil {
+			return nil, serializer.NewHTTPError(http.StatusBadRequest, err.Error())
+		}
+
+		nodes, err := service.UnmarshallUAST(&data)
+		if err != nil {
+			return nil, serializer.NewHTTPError(http.StatusBadRequest, err.Error())
+		}
+
+		resp := &uast.Node{InternalType: "Search results"}
+
+		if req.Filter != "" {
```

Comment:
Maybe an empty filter could be a bad request error, to simplify this part of the code.

Reply:
I'm open to this change. The only problem is then we will move complexity to frontend. Please let me know it's better to have it in JS than in Go.

Reply:
If you are talking about disabling the search button when the input is empty, I think it adds a nice feel to the interface and it's not that much complexity. I would make this change, but I'm also OK merging as it is. Up to you :+1:

Reply:
I'm talking about how to return to full UAST.
Right now you can filter than just remove the filter query and get origin UAST back.
If we disable search button we would need one more button for it :) I personally use this feature.

Reply:
I think for now I would keep current behavior to be consistent with dashboard. But feel free to create an issue if you see how to improve it! Thank you.